File: longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s;
        int count = 0;
        for(auto x: nums){
            s.insert(x);
        }
        for(auto x : nums){
            if(s.find(x-1) == s.end()){
                int length = 1;
                int num = x + 1;
                while(s.find(num) != s.end()){
                    length++;
                    num++;
                }
                count = max (count, length);
            }
        }
        return count;
    }
};

File: longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        
        Arrays.sort(nums);
        int maxLen = 1;
        int count = 1;
        
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i - 1] + 1) {
                count++;
            } else if (nums[i] != nums[i - 1]) {
                maxLen = Math.max(maxLen, count);
                count = 1;
            }
        }
        
        return Math.max(maxLen, count);
    }
}